src/org/rascalmpl/library/lang/json/internal/JsonValueReader.java

```diff
@@ -269,26 +269,30 @@ private IValue sourceLocationObject() throws IOException {
         }
         
         in.endObject();
-
-        if (path != null && offset != -1 && length != -1 && beginLine != -1 && endLine != -1 && beginColumn != -1 && endColumn != -1) {
-            return vf.sourceLocation(path, offset, length, beginLine, endLine, beginColumn, endColumn);
```

Comment:
this code would create file locations, also for things like `|project://x/p|(...)`. So I rewrote it to not use the more problematic constructor, and properly support offsets for non-file based locs.

Reply:
good find. this would have exploded at some time in the future